src/daemon/daemon.cpp

```diff
@@ -2662,11 +2663,26 @@ error_string mp::Daemon::async_wait_for_ssh_and_start_mounts_for(const std::stri
                 }
                 catch (const std::exception& e)
                 {
-                    fmt::format_to(errors, "Removing \"{}\": {}\n", target_path, e.what());
+                    // TODO: Combine these into one warning level log once they are displayed in the cli by
+                    // default
+                    mpl::log(mpl::Level::info, category, fmt::format("Removing \'{}\': {}\n", target_path, e.what()));
+                    fmt::format_to(warnings,
+                                   fmt::format("Removing mount \'{}\' from {}: {}\n", target_path, name, e.what()));
                     invalid_mounts.push_back(target_path);
                 }
-                persist_instances();
             }
+
+            for (const auto& mount : invalid_mounts)
+                vm_instance_specs[name].mounts.erase(mount);
+
+            if (server && !fmt::to_string(warnings).empty())
+            {
+                Reply reply;
+                reply.set_log_line(fmt::to_string(warnings));
```

Comment:
this is calling `fmt::to_string` twice, which involves two string allocations. you could use `warnings.size() > 0` instead